ViewEnvironmentUI/Sources/ViewEnvironmentPropagating.swift

```diff
@@ -432,3 +462,62 @@ extension ViewEnvironmentPropagating {
 }
 
 private class ViewEnvironmentUpdateObservationKey: NSObject {}
+
+/// A closure that customizes the `ViewEnvironment` as it flows through a propagation node.
+///
+public typealias ViewEnvironmentCustomization = (inout ViewEnvironment) -> Void
+
+/// Describes the lifetime of a `ViewEnvironment` customization.
+///
+/// This customization will be removed when `remove()` is called or the lifetime token is
+/// de-initialized.
+///
+/// ## SeeAlso ##
+/// - `addEnvironmentCustomization(_:)`
+///
+public final class ViewEnvironmentCustomizationLifetime {
+    /// Removes the observation.
+    ///
+    /// This is called in `deinit`.
+    ///
+    public func remove() {
+        onRemove()
+    }
+
+    private let onRemove: () -> Void
```

Comment:
In `TrampolineModalPresenter.LifetimeToken` we nil out the callback after the first call, and assert if you try to manually dismiss twice. What do you think about doing that here too?

Reply:
I don't have a strong take—I'm fine with adding it!

- [ ] assert `remove()` is only called at most once